### PR TITLE
Add Cylc UI Server

### PR DESCRIPTION
**Pull request by kinow**
PR to add Cylc UI Server, based on [a recipe](https://github.com/cylc/cylc-conda/tree/c4b692f5d9ba949b4005b1f390c498df3b62e758/recipes/cylc-uiserver) that we used while releasing our first alpha version to a private conda channel. Only tweaked the requirements to remove unnecessary ones from `requirements|host`.

The requirement on `pytest-runner=4.4` under the `host` section is legit, but will fix it in a future release of `cylc-uiserver`, and will update the feedstock later too :+1: 

Tested locally with `conda build .` && `conda install --use-local cylc-uiserver`, then started the application successfully.

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Thanks!
Bruno

### CODE REVIEW

**Comment by kinow** [comment]
Trying to understand how to solve what seems to be a dependency mismatch somewhere in our dependency tree for this recipe.

```bash
Package python conflicts for:
tornado[version='>=6.0'] -> python[version='>=3.7,<3.8.0a0']
python[version='>=3.6,<3.7.0a0']
graphene-tornado[version='>=2.1,<2.6'] -> graphene[version='>=2.1,<3'] -> graphql-relay[version='>=0.4.5,<1,>=2,<3'] -> graphql-core[version='>=0.5.0,>=2.2,<3'] -> promise[version='>=2.1'] -> typing -> python[version='3.6.*,>=3.6,<3.7.0a0']
jupyterhub[version='>=1.0'] -> prometheus_client[version='>=0.0.21'] -> twisted -> service_identity[version='>=18.1.0'] -> attrs[version='>=16.0.0'] -> hypothesis -> coverage -> python[version='3.6.*,>=3.6,<3.7.0a0']
Package typing conflicts for:
cylc-flow==8.0a1 -> graphene==2.1.6 -> graphql-relay[version='>=0.4.5,<1'] -> graphql-core[version='>=0.5.0,>=2.2,<3'] -> promise[version='>=2.1'] -> typing
##[error]Bash exited with code '1'.
```

**Comment by kinow** [comment]
Changed the Python requirement from `>=3.7` to `>=3.5` 👍

File: recipes/cylc-uiserver/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.5
+    - pip
+    # TODO: build failed with pytest-runner only, because the setup.py of cylc-uiserver specifies this
+    # version of pytest-runner. Need to make pytest-runner a dependency only for python setup.py test
+    # and then remove this from requirements|host in the recipe
+    - pytest-runner ==4.4
```
Comment (marcelotrevisani) [inline recipes/cylc-uiserver/meta.yaml]: Just a small hint here, the ``==`` is not necessary :)
Not necessary to change, but you can just write
```suggestion
    - pytest-runner  4.4
```

File: recipes/cylc-uiserver/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+build:
+  number: 0
+  # cylc-uiserver provides JupyterHub extensions and a small Jupyter Application. As JupyterHub
+  # does not run on Windows, we skip the build for that platform here
+  skip: True  # [not linux]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.5
```
Comment (marcelotrevisani) [inline recipes/cylc-uiserver/meta.yaml]: As this recipe is not ``noarch: python`` you will need to skip those python versions on ``build``

File: recipes/cylc-uiserver/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 4cc85a6680148cddd8f1fd714f772acb7c52487239b7c2b781bf543a51c099d1
+
+build:
+  number: 0
+  # cylc-uiserver provides JupyterHub extensions and a small Jupyter Application. As JupyterHub
+  # does not run on Windows, we skip the build for that platform here
+  skip: True  # [not linux]
```
Comment (marcelotrevisani) [inline recipes/cylc-uiserver/meta.yaml]:
```suggestion
  skip: True  # [not linux]
  skip: True  # [py2k]
```

File: recipes/cylc-uiserver/meta.yaml (hunk 2 above)
Comment (marcelotrevisani) [reply recipes/cylc-uiserver/meta.yaml]:
```suggestion
    - python
```

File: recipes/cylc-uiserver/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+requirements:
+  host:
+    - python >=3.5
+    - pip
+    # TODO: build failed with pytest-runner only, because the setup.py of cylc-uiserver specifies this
+    # version of pytest-runner. Need to make pytest-runner a dependency only for python setup.py test
+    # and then remove this from requirements|host in the recipe
+    - pytest-runner ==4.4
+  run:
+    - python >=3.5
```
Comment (marcelotrevisani) [inline recipes/cylc-uiserver/meta.yaml]:
```suggestion
    - python
```

File: recipes/cylc-uiserver/meta.yaml (hunk 1 above)
Comment (kinow) [reply recipes/cylc-uiserver/meta.yaml]: Oh that works too? Good to know, looms simpler that way. Thanks!

**Comment by kinow** [comment]
Thanks @marcelotrevisani !

**Comment by kinow** [comment]
Cylc Flow uses Py3.7 features. Its recipe is already skipping <3.6, and it was missing on this one :+1: build passing, ready for review. Thanks!

**Comment by scopatz** [comment]
Thanks!

**Comment by kinow** [comment]
Thank you @scopatz !

**Pull request merged**